mautic/mautic
# Integration sync flag

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ yes]
| New feature/enhancement? (use the a.x branch)      | [4.x ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ yes ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This solution is only effective for new enabled plugins, or you have edit and save the plugin settings.

There are plugins that only pull from the service through the integration sync framework but doesn't push. This has exposed a missed design flaw that assumes all integrations will sync both ways.

The result is that the integration framework tracks a field change per field per contact for these integrations that never gets purged causing the mautic_sync_object_field_change_report table to fill up with useless data that could impact performance of other integrations that sync both ways.

#### Steps to reproduce
Enable a integration plugin and save mapped fields with only mautic (one way(import)) sync.
Try to change the synced contact/company
you will see an entry in DB table 'mautic_sync_object_field_change_report', Ideally this entry would be cleared once it syncs back to integration. Since it is only mautic sync, the entry is being ignored.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Load up [this PR]
3. Enable plugin (Salesforce)
4. Add mapping fields with only mautic integration
5. Once the contacts/company details are synced.
6. Try to edit the contact/company
7. There shouldn't be any entry to 'mautic_sync_object_field_change_report'
